### PR TITLE
T28 : Queue<T>

### PR DESCRIPTION
**Pull request by RaaneshKumar**

### CODE REVIEW

File: Test/UnitTest1.cs [hunk 1]
```diff
@@ -1,8 +1,39 @@
...
+      public void PeekTest () {
+         Assert.ThrowsException<InvalidOperationException> (() => myQueue.Peek ());
+         myQueue.Enqueue (1);
+         myQueue.Enqueue (2);
+         queue.Enqueue (1);
+         queue.Enqueue (2);
+         Assert.AreEqual (queue.Peek (), myQueue.Peek ());
+      }
+
+      TQueue<int> myQueue = new ();
```
Comment (priyarathna) [inline Test/UnitTest1.cs]: Create only one queue and check the queue data with hard-coded values. For eg: Assert.AreEqual (queue.Count, 2);

File: Test/UnitTest1.cs [hunk 2]
```diff
@@ -1,8 +1,39 @@
...
+      Queue<int> queue = new ();
```
Comment (priyarathna) [inline Test/UnitTest1.cs]: Local variable must start with 'm'. Apply this wherever necessary.

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: Test/UnitTest1.cs [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+
+      [TestMethod]
+      public void PeekTest () {
+         Assert.ThrowsException<InvalidOperationException> (() => mQueue.Peek ());
+         mQueue.Enqueue (1);
+         mQueue.Enqueue (2);
+         Assert.AreEqual (1, mQueue.Peek ());
+      }
+
+      TQueue<int> mQueue = new ();
```
Comment (priyarathna) [inline Test/UnitTest1.cs]: 'mQueue' is not initialized for each test. Are all tests getting passed?

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: Test/UnitTest1.cs (hunk 3 above)
Comment (RaaneshKumar) [reply Test/UnitTest1.cs]: 'mQueue' is initialized outside in common and accessed in all test methods. Yes ma'am, all the tests are getting passed.
Should I initialize it for each test or is this ok?

File: Program.cs [hunk 4]
```diff
@@ -3,17 +3,65 @@
 // Copyright (c) Metamation India.                                                
 // ---------------------------------------------------------------------------------------
 // Program.cs                                                                     
-// 
-// 
-// 
+// Program to create a custom TQueue<T> class.
```
Comment (priyarathna) [inline Program.cs:6]: There is no 'csproj' file. Make sure whether you've pushed all the necessary files.

**Review by priyarathna — CHANGES_REQUESTED** [review]

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**